cmake/Helpers.cmake

```diff
@@ -0,0 +1,790 @@
+FUNCTION(SET_INTERNAL VAR_NAME VALUE)
+    SET(${VAR_NAME} ${VALUE} CACHE INTERNAL "")
+ENDFUNCTION()
+
+FUNCTION(ADD_FLAG_IF_SUPPORTED)
+    SET(options "")
+    SET(oneValueArgs FLAG TEST_VALUE VALUE)
+    SET(multiValueArgs "")
+    CMAKE_PARSE_ARGUMENTS("" "${options}" "${oneValueArgs}" "${multiValueArgs}" ${ARGN})
+
+    IF(${_TEST_VALUE})
+        SET(TEST_FLAG "${_FLAG}=${_TEST_VALUE}")
+    ELSEIF(${_VALUE})
+    SET(TEST_FLAG "${_FLAG}=${_VALUE}")
+    ELSE ()
+        SET(TEST_FLAG "${_FLAG}")
+    ENDIF ()
+    
+    CHECK_CXX_COMPILER_FLAG("${TEST_FLAG}" SUPPORTED)
+
+    IF(SUPPORTED)
+        IF(${_VALUE})
+            SET(_FLAG " ${_FLAG}=${_VALUE}")
+        ENDIF ()
+        STRING(APPEND CMAKE_CXX_FLAGS "${_FLAG}")
```

Comment:
Pretty sure this is missing a space.

Reply:
```
    IF(SUPPORTED)
        IF(${_VALUE})
            SET(_FLAG " ${_FLAG}=${_VALUE}")
        ENDIF ()
        STRING(APPEND CMAKE_CXX_FLAGS "${_FLAG}")
    ENDIF ()
 ```
 Which line exactly?

Reply:
The one I commented under. If ${_VALUE} does not evaluate to true, what will ensure _FLAG has a space in front?

Reply:
Ok I thought it was about indentation. I get it now :D